src/libstd/collections/hash/table.rs

```diff
@@ -835,18 +838,27 @@ pub struct IterMut<'a, K: 'a, V: 'a> {
     elems_left: usize,
 }
 
+unsafe impl<'a, K: Sync, V: Sync> Sync for IterMut<'a, K, V> {}
+unsafe impl<'a, K: Send, V: Send> Send for IterMut<'a, K, V> {}
```

Comment:
Actually, after thinking some more, this isn't so obvious. The implementation is `Iterator<Item = (&K, &mut V)>`, so "naively" the `K` requires `Sync`, however, in practice the `&K` is guaranteed to be unaliased (it's "really" `&uniq K`) so maybe `K: Send` works here.

I'm a bit confused about the correctness here. I'm now thinking that either `K: Sync` or `K: Send` are correct, but that seems mildly strange to me. I'll ask.